backend/eventd/entity.go

```diff
@@ -45,10 +45,11 @@ func createProxyEntity(event *corev2.Event, s store.EntityStore) error {
 			entity.EntityClass = corev2.EntityProxyClass
 			entity.Subscriptions = append(entity.Subscriptions, corev2.GetEntitySubscription(entityName))
 		}
+	}
 
-		if err := s.UpdateEntity(ctx, entity); err != nil {
-			return err
-		}
+	entity.CreatedBy = event.CreatedBy
+	if err := s.UpdateEntity(ctx, entity); err != nil {
+		return err
```

Comment:
Why is this now outside the if statement? This will cause proxy entities to be written to every time an event fires, probably not what we want.

Reply:
Otherwise, existing proxy entities will not get updated with the correct `created_by` field.

Reply:
We could check to see if `created_by` has changed, and only update if that's the case. Probably more performant that way.

Reply:
I don't think that's quite right either. It makes sense for the case where proxy entities are being created for the first time, but if Alice creates a proxy entity, and then Bob creates an event referencing one, Alice is still the creator of the proxy entity.

Reply:
Bob's event could still modify the proxy entity that Alice originally created though?

Reply:
I think the only time Bob's event should result in a proxy entity that Bob owns is when there wasn't one before, and it triggers the automated proxy entity creation. Normally, the entity wouldn't be touched by this process.

Reply:
So if Bob were to update metadata labels of an existing proxy entity via the [events API](https://docs.sensu.io/sensu-go/latest/api/events/#events-post), then the desired behavior would be for those changes _not_ to be persisted?

Reply:
Well, desire is in the eye of the user, but Sensu's behaviour up until now has been not to modify entities when processing events, except for keepalives.

Reply:
Seems a little strange that's the case, but will revert those changes as to not change the behavior.

Reply:
I do think created_by should be set if the proxy entity is created however.